proposals/distribution.md

```diff
@@ -57,10 +57,90 @@ The API spec is currently considered v2 and we will start the specification at v
+
+The following entries should be added to the [scope table][scope]:
+
+* “Specifying authentication and authorization schemes”.
+    Docker's current registry uses an [extension][token] of the [`Bearer`][rfc6750] [auth scheme][rfc7235-s2.1].
+    Work on specifying Docker's scheme will continue independently, and is orthogonal to the registry API.
+
+    * What: Specifying authentication and authorization schemes
+    * In/Out/Future: Out of scope
+    * Status: N/A
+    * Description: Defining protocols for authenticating and authorizing distribution access.
+    * Why: As a HTTP-based protocol, clients and servers can negotiate authentication via HTTP's [challenge-response authentication framework][rfc7235-s2.1].
+        There is no need for the distribution specificaiton to choose a particular authentication scheme, because clients receiving 401 and 407 responses can use IANA's [HTTP Authentication Scheme Registry][iana-auth] to look up referenced schemes and take appropriate action.
+frameworkIt is reasonable to provide a standardized way to use DNS based distribution in conjunction with OCI without requiring its use.
+
+* “Creating a reference spec for optional DNS based naming and discovery”.
+    Discovery and registry protocols are completely separate and do not need to be added together.
+    This entry replaces part of the previous “Creating Reference spec for optional DNS based naming & distribution” entry.
+
+    * What: Creating a reference spec for optional DNS based naming and discovery
+    * In/Out/Future: In scope for future specification
+    * Status: Not currently being worked
+    * Description: Define a protocol for resolving an image name to retrieval information.
+        When we address this, we will also allow for alternative, parallel name-to-image discovery protocols in parallel with the OCI-specified protocol.
+    * Why: It is reasonable to provide a standardized way to use DNS based distribution in conjunction with OCI without requiring its use.
+        There are many good use cases for DNS based distribution, but not all use cases support this.
+        Furthermore, encoding the location of a bundle into the bundle can cause issues with downloads from alternate locations other than the origin specified in the name.
+
+* “Specifying a distribution method”.
+    This entry replaces part of the previous “Creating Reference spec for optional DNS based naming & distribution” and “Standardizing on a particular Distribution method” entries.
+
+    Retrieving images covers the current “tag listing”, because tags are entries in the image format's manifests array.
+    Other tag-listing endpoints needed for backwards-compatibility are therefore in scope as well.
+
+    Repository actions and listing images within a repository are considered part of distribution policy or content management and are out of scope for this entry's per-image action.
```

Comment:
We should discuss this.  If I can't list or at least search the contents of a repository.. how do I find out what images are in the repository? Is there some valid pattern of use of an image repository without listing and search of the contents within? At the minimum I'd like to see such issues as open for discussion vs explicitly listed as out of scope.

Reply:
> We should discuss this.

Agreed.  Previous discussion starting [here][1] and [here][2], but pinning this sort of thing down is what this helper PR is about.

> If I can't list or at least search the contents of a repository.. how do I find out what images are in the repository?

I think that use-case (like listing/adding/removing repos) is part of a separate content-management API.  This distribution spec, on the other hand, is only scoped for per-image actions.  For example, how to pull a particular image from a registry is in scope, but discovering  an image name to pull is not.

[1]: https://github.com/opencontainers/tob/pull/35#discussion_r163485706
[2]: https://github.com/opencontainers/tob/pull/37#issuecomment-360888967

Reply:
It sounds like you are saying I should be able to get a full manifest list for something explicitly requested eg. namespace "library/" image name "busybox". But listing / searching for "busybox" irregardless of namespace would be out of scope. Let's at least separate the issues a bit more. Content management seems to be to broadly defined here. Managing lists of repos should not be in the same scope as managing manifest lists in a repo.

Reply:
> It sounds like you are saying I should be able to get a full manifest list for something explicitly requested eg. namespace "library/" image name "busybox". But listing / searching for "busybox" irregardless of namespace would be out of scope.

Yes.

> Let's at least separate the issues a bit more. Content management seems to be to broadly defined here.

The per-issue scoping is my guess based on [these][1] [comments][2].

> Managing lists of repos should not be in the same scope as managing manifest lists in a repo.

I'm fine with splitting them up into separate levels, but I don't see why you'd need to be able to list images in a repository to be able to push/pull a given image/manifest.  Listing tags/manifests within an image, on the other hand, is in scope with my current wording.  Can you provide more detail about how you'd like to see that adjusted?

[1]: https://github.com/opencontainers/tob/pull/37#issuecomment-360968873
[2]: https://github.com/opencontainers/tob/pull/37#issuecomment-360908827
[3]: https://github.com/opencontainers/tob/pull/37/commits/5cefabcae103c79a4783191a575f50c510c28832#diff-748aea9c83a752af811a80671d8803d2R107

Reply:
> I'm fine with splitting them up into separate levels, but I don't see why you'd need to be able to list images in a repository to be able to push/pull a given image/manifest.

I guess the point is you might not know what the namespace is for a "given" image, "library/" only works if it's in the default namespace. Other points include, "how do you know what you want if you don't know what's in there." 

> Listing tags/manifests within an image, on the other hand, is in scope with my current wording. Can you provide more detail about how you'd like to see that adjusted?

I think your wording might be ok, I just wanted to get clarification that I read it correctly. Feel free to add my example. Maybe @stevvooe will inject some comments.

Reply:
> I guess the point is you might not know what the namespace is for a "given" image, "library/" only works if it's in the default namespace.

Right.  I'd consider `library/busybox` to be the image, not just the unqualified `busybox`.

> Other points include, "how do you know what you want if you don't know what's in there."

This is discovery.  It's important, but [orthogonal and out of scope for this project][1].

> I think your wording might be ok, I just wanted to get clarification that I read it correctly.  Feel free to add my example.

Ok, I've pushed 28a8ab3 → 14ad993 incorporating your `library/busybox` example.

[1]: https://github.com/opencontainers/tob/issues/34#issuecomment-350529321